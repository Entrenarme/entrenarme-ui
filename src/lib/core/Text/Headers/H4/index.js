// @flow
import * as React from 'react';
import styled from 'styled-components';

import * as fonts from '../../../../helpers/fonts';
import colors from '../../../../helpers/colors';
import { getFontWeight } from '../utils';

const normalDesktop = 1.25;

const normalMobile = 1.125;

const sizeModifier = 0.25;

const textSizeTransform = (size, device) => {
  if (size === 'large') {
    return device === 'desktop'
      ? normalDesktop + sizeModifier
      : normalMobile + sizeModifier;
  } else if (size === 'small') {
    return device === 'desktop'
      ? normalDesktop - sizeModifier
      : normalMobile - sizeModifier;
  } else {
    return device === 'desktop' ? normalDesktop : normalMobile;
  }
};

const marginBottomSize = (size, device, nomargin) => {
  if (nomargin) return 0;
  if (size === 'large') {
    return device === 'desktop' ? 1.875 : 1.25;
  } else if (size === 'small') {
    return device === 'desktop' ? 0.625 : 0.3125;
  } else {
    return device === 'desktop' ? 1.25 : 0.625;
  }
};

const StyledH4 = styled.h4`
  text-transform: uppercase;
  font-size: ${props =>
    `${textSizeTransform(props.options.size, 'desktop')}rem`};
  font-weight: ${props => getFontWeight(props)};
  font-family: ${fonts.mainFont};
  text-align: ${props => props.options.textAlign};
  color: ${props => colors[props.options.color].default};
  margin-top: 0rem;
  margin-left: 0rem;
  margin-right: 0rem;
  margin-bottom: ${props =>
    `${marginBottomSize(
      props.options.size,
      'desktop',
      props.options.nomargin,
    )}rem`};

  /* Smartphones (portrait) ----------- */
  @media only screen and (max-width: 320px) {
    font-size: ${props =>
      `${textSizeTransform(
        props.options.size,
        'mobile',
        props.options.nomargin,
      )}rem`};
    margin-bottom: ${props =>
      `${marginBottomSize(
        props.options.size,
        'mobile',
        props.options.nomargin,
      )}rem`};
  }

  /* iPhone 5/6/6+ ----------- */
  @media only screen and (min-width: 320px) and (max-width: 767px) {
    font-size: ${props =>
      `${textSizeTransform(props.options.size, 'mobile')}rem`};
    margin-bottom: ${props =>
      `${marginBottomSize(
        props.options.size,
        'mobile',
        props.options.nomargin,
      )}rem`};
  }

  /* iPads (portrait and landscape) ----------- */
  @media only screen and (min-width: 768px) and (max-width: 1223px) {
    /* Styles */
  }

  /* Desktops and laptops ----------- */
  @media only screen and (min-width: 1224px) {
    /* Styles */
  }

  /* Large screens ----------- */
  @media only screen and (min-width: 1824px) {
    /* Styles */
  }

  /* ----------- Retina Screens ----------- */
  @media screen and (min-device-width: 1200px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
    /* Styles */
  }
`;

type Props = {
  children: React.Node,
  color?: color,
  size?: 'small' | 'medium' | 'large',
  textAlign?: 'left' | 'center' | 'right',
  nomargin: boolean,
  fontWeight: 'regular' | 'bold' | 'bolder' | 'boldest',
};

const H4 = ({
  children,
  color,
  size,
  textAlign,
  nomargin,
  fontWeight,
  ...rest
}: Props) => (
  <StyledH4
    {...rest}
    options={{ color, size, textAlign, nomargin, fontWeight }}
  >
    {children}
  </StyledH4>
);

H4.defaultProps = {
  color: 'primary',
  size: 'medium',
  textAlign: 'left',
  nomargin: false,
  fontWeight: 'regular',
};

export default H4;
