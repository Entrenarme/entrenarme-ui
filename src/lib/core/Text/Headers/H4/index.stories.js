// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';

import H4 from './index';
import P from '../../Paragraph/index';

storiesOf('Text/Headers/H4', module).add('default', () => (
  <div>
    <H4>Heading 4</H4>
    <P>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum
      magna eu velit semper hendrerit. Aenean aliquet pretium justo, ut rhoncus
      magna euismod et. Donec condimentum congue sapien, nec auctor eros
      sagittis eget. Phasellus eleifend risus et arcu laoreet, vitae lobortis
      turpis feugiat. Maecenas pulvinar ultricies risus nec tincidunt. Donec
      tristique imperdiet dui at egestas. Praesent ornare et ante sit amet
      condimentum.
    </P>
  </div>
));

storiesOf('Text/Headers/H4', module).add('small', () => (
  <div>
    <H4 size="small">Heading 4</H4>
    <P>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum
      magna eu velit semper hendrerit. Aenean aliquet pretium justo, ut rhoncus
      magna euismod et. Donec condimentum congue sapien, nec auctor eros
      sagittis eget. Phasellus eleifend risus et arcu laoreet, vitae lobortis
      turpis feugiat. Maecenas pulvinar ultricies risus nec tincidunt. Donec
      tristique imperdiet dui at egestas. Praesent ornare et ante sit amet
      condimentum.
    </P>
  </div>
));

storiesOf('Text/Headers/H4', module).add('large', () => (
  <div>
    <H4 size="large">Heading 4</H4>
    <P>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum
      magna eu velit semper hendrerit. Aenean aliquet pretium justo, ut rhoncus
      magna euismod et. Donec condimentum congue sapien, nec auctor eros
      sagittis eget. Phasellus eleifend risus et arcu laoreet, vitae lobortis
      turpis feugiat. Maecenas pulvinar ultricies risus nec tincidunt. Donec
      tristique imperdiet dui at egestas. Praesent ornare et ante sit amet
      condimentum.
    </P>
  </div>
));

storiesOf('Text/Headers/H4', module).add('color warning', () => (
  <div>
    <H4 color="warning">Heading 4</H4>
    <P>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum
      magna eu velit semper hendrerit. Aenean aliquet pretium justo, ut rhoncus
      magna euismod et. Donec condimentum congue sapien, nec auctor eros
      sagittis eget. Phasellus eleifend risus et arcu laoreet, vitae lobortis
      turpis feugiat. Maecenas pulvinar ultricies risus nec tincidunt. Donec
      tristique imperdiet dui at egestas. Praesent ornare et ante sit amet
      condimentum.
    </P>
  </div>
));
