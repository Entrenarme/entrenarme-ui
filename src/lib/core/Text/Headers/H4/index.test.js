import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';

import H4 from './index';

test('renders "Heading" with no props', () => {
  const { container } = render(<H4>Heading</H4>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with large size', () => {
  const { container } = render(<H4 size="large">Heading</H4>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with small size', () => {
  const { container } = render(<H4 size="small">Heading</H4>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with color prop', () => {
  const { container } = render(<H4 color="warning">Heading</H4>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders "Heading" with text align', () => {
  const { container } = render(<H4 textAlign="center">Heading</H4>);
  expect(container.firstChild).toMatchSnapshot();
});
