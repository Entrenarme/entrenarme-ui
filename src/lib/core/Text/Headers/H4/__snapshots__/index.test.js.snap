// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Heading" with color prop 1`] = `
<h4
  class="sc-bdVaJa juADiy"
>
  Heading
</h4>
`;

exports[`renders "Heading" with large size 1`] = `
<h4
  class="sc-bdVaJa dwKxth"
>
  Heading
</h4>
`;

exports[`renders "Heading" with no props 1`] = `
<h4
  class="sc-bdVaJa iaqJBm"
>
  Heading
</h4>
`;

exports[`renders "Heading" with small size 1`] = `
<h4
  class="sc-bdVaJa iRxabv"
>
  Heading
</h4>
`;

exports[`renders "Heading" with text align 1`] = `
<h4
  class="sc-bdVaJa cFgHgH"
>
  Heading
</h4>
`;
